go-controller/pkg/ovn/egressfirewall_dns.go

```diff
@@ -0,0 +1,194 @@
+package ovn
+
+import (
+	"fmt"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/klog"
+)
+
+type EgressDNS struct {
+	// Protects pdMap/namespaces operations
+	lock sync.Mutex
+	// holds DNS entries globally
+	dns *util.DNS
+	// this map holds dnsNames to the dnsEntries
+	dnsEntries map[string]*dnsEntry
+	// allows for the creation of addresssets
+	addressSetFactory AddressSetFactory
+
+	// Report change when Add operation is done
+	added          chan string
+	stopChan       chan struct{}
+	controllerStop <-chan struct{}
+}
+
+type dnsEntry struct {
+	//this map holds all the namespaces that a dnsName appears in
+	namespaces map[string]struct{}
+	// the current IP addresses the dnsName resolves to
+	dnsResolves []net.IP
+	// the addressSet that contains the current IPs
```

Comment:
whats the difference between dnsResolves and dnsAddressSet? Are they both holding the same data, just in different formats?

Reply:
I had to use it for testing. It holds the same values but there is no way to access the IP addresses in an address_set outside of the address_set package. I could instead of mocking the address_set code try to use the fake address_set used elsewhere

Reply:
ah ok. maybe a comment that says for testing purposes or something would make this clearer but nbd